src/WinRT.Interop.Generator/Extensions/WindowsRuntimeExtensions.cs

```diff
@@ -48,6 +81,40 @@ public bool IsProjectedWindowsRuntimeClassType
             }
         }
 
+        /// <summary>
+        /// Checks whether a <see cref="TypeDefinition"/> represents a type that can be constructed (i.e. instantiated).
+        /// </summary>
+        public bool IsConstructibleType => type is { IsInterface: false, IsAbstract: false };
+
+        /// <summary>
+        /// Checks whether a <see cref="TypeDefinition"/> represents a type that can possibly be exposed to Windows Runtime.
+        /// </summary>
+        public bool IsPossiblyWindowsRuntimeExposedType
```

Comment:
How is this used?

Reply:
It's used when discovering user-defined types that should be exposed, to quickly filter out types that can't possibly be exposed (e.g. static types, ref struct types, etc.). It's just to avoid the more expensive work in those scenarios.